forc-plugins/forc-doc/src/main.rs

```diff
@@ -77,8 +80,78 @@ pub fn main() -> Result<()> {
     }
     fs::create_dir_all(&doc_path)?;
 
-    // build core documentation
-    build_docs(&manifest, pkg_manifest, &doc_path, &build_instructions)?;
+    println!(
+        "   {} {} ({})",
+        "Compiling".bold().yellow(),
+        pkg_manifest.project_name(),
+        manifest.dir().to_string_lossy()
+    );
+
+    let member_manifests = manifest.member_manifests()?;
+    let lock_path = manifest.lock_path()?;
+    let plan = pkg::BuildPlan::from_lock_and_manifests(
+        &lock_path,
+        &member_manifests,
+        build_instructions.locked,
+        build_instructions.offline,
+    )?;
+
+    let type_engine = TypeEngine::default();
+    let decl_engine = DeclEngine::default();
+    let query_engine = QueryEngine::default();
+    let engines = Engines::new(&type_engine, &decl_engine, &query_engine);
+    let tests_enabled = build_instructions.document_private_items;
+    let mut compile_results = pkg::check(
+        &plan,
+        BuildTarget::default(),
+        build_instructions.silent,
+        tests_enabled,
+        engines,
+    )?;
+
+    if !build_instructions.no_deps {
+        let order = plan.compilation_order();
+        let graph = plan.graph();
+        let manifest_map = plan.manifest_map();
+
+        for (node, compile_result) in order.iter().zip(compile_results) {
+            let id = &graph[*node].id();
+
+            if let Some(pkg_manifest_file) = manifest_map.get(id) {
+                let manifest_file = ManifestFile::from_dir(pkg_manifest_file.path())?;
+                let ty_program = match compile_result.value.and_then(|programs| programs.typed) {
+                    Some(ty_program) => ty_program,
+                    _ => bail!("CompileResult returned None"),
```

Comment:
I would adjust both `bail` messages for two reasons:

1. They are the same so it's unclear which part of the code triggers the error
2. It's unclear to me what it means for a `CompilationResult` to return `None`. I would change it to be more descriptive to indicate what happened

Reply:
That makes sense, will do.